src/target/riscv/riscv.c

```diff
@@ -2390,23 +2396,28 @@ static int riscv_virt2phys(struct target *target, target_addr_t virtual, target_
 	int enabled;
 	if (riscv_mmu(target, &enabled) != ERROR_OK)
 		return ERROR_FAIL;
-	if (!enabled)
-		return ERROR_FAIL;
-
+	if (!enabled) {
+		*physical = virtual;
```

Comment:
Does this fix anything, or is this just defensive coding?

Reply:
As you can see, logic of `read/write_memory` functions  was changed (assignment to `address` was removed). So, when MMU is disabled we want to perform a memory access using untraslated address, and for this we need this change.

Reply:
Thanks for explaining.